### ISSUE TITLE
Provider Self-Input Form

### ISSUE THREAD

**Issue by rileydjohnson** [radremedy/radremedy#253]
Adding the provider form under "Submit a Provider"

Have a landing page with "Are you a provider?" and "Are you an individual?" that goes to the two different forms?

Provider form - http://goo.gl/forms/4GX5vBO587

Also make sure it is fixed in the FAQ

**Comment by lovecultivated** [comment]
This is ready to go live

**Comment by radioprotector** [comment]
@internetcreep: Here's what I have so far.

## Functional Overview

### Submitting Resources

We'll have a form with the following fields in this order:

- Provider Information
	* Provider Name (required)
	* Organization
	* Description
	* Address
	* Phone Number
	* Fax
	* Email
	* Website
	* Office Hours
	* Categories (using the standard multiselect)
	* Populations Served (using the standard multiselect)
	* Hospital Affiliation
	* NPI
	* Other Notes
- Review
	* Rating (required)
	* Intake Rating
	* Staff Rating
	* Review Text (required)

Behind the scenes, we'll also track:
- Submitted IP
- Submitted User
- Source (will go in as "`user - <username>`")

We'll make the following options always available even if the current user isn't logged in, but update the login process to redirect them to their requested page upon a successful login:

- Submit a Provider
- Add a Review
- Submit a Correction

All submitted resources will default to active but not approved. Approving a resource will toggle the approved flag, but rejecting a resource will change the status to inactive.

From an Admin perspective, we'll have a "Pending Submissions" section under the "Resource" option. From there, each individual resource will have "Save", "Approve", "Reject", and "Cancel" options. Resources can be deleted from this view but not created.

I can look into trying to get some duplicate detection in but I fear that's going to be pretty difficult to manage.

### Searching Resources

We will update all standard resource retrievals to filter on having a status of approved. Attempting to go to a single unapproved resource will result in "Not Found".

### Future Steps

I think we need to start separating out the "Settings" page on the user profile. Instead of linking to the "Settings" from the main layout page when someone's logged in, we should instead link to a new "Profile" option with the following sections:

- Profile details (Essentially the "My Profile" part that's currently there), with a link to edit them
- My Reviews (Showing only visible top-level reviews on active/approved providers)
- My Submitted Resources
- Change My Password link

This probably ties in to some of the work outlined in #250 a bit.

**Comment by radioprotector** [comment]
Never mind - wrong issue. Will comment on the appropriate one.

**Comment by alison985** [comment]
I started on this. https://github.com/alison985/radremedy/commit/8416d294c97340e6ae10a9078a1f263d14063f58

**Comment by radioprotector** [comment]
Looks good so far!
@lovecultivated: do you have any feedback on fields we want that aren't in the Google form that we want in RAD, or vice versa?

**Comment by lovecultivated** [comment]
I am assuming that this is the provider self-input form?

The google form is what we are working off of...  @alison985
<https://github.com/alison985/radremedy/commit/8416d294c97340e6ae10a9078a1f263d14063f58>
would
a pdf copy of that me helpful?

Rachel A. Hennessy
*Developmental Therapist*
*(773) 817-8044*

IMPORTANT: This electronic transmission contains confidential information,
some or all of which may be protected health information as defined by the
federal Health Insurance Portability & Accountability Act (HIPAA) Privacy
Rule. This transmission is intended for the exclusive use of the individual
or entity to which it is addressed and may contain information that is
proprietary, privileged, confidential and/ or exempt from disclosure under
applicable law.



If you are not the intended recipient (or an employee or agent responsible
for delivering this electronic transmission to the intended recipient), you
are herby notified that any disclosure, dissemination, distribution, or
copying if this information is strictly prohibited and may be subject to
legal restriction or sanction. Please notify the sender by telephone
(number listed above) to arrange the return or destruction of the
information and all copies.

**Comment by alison985** [comment]
@lovecultivated Yes, a PDF copy of the form would be very helpful so I don't have to submit bad data to the first page of the form to get to the second.

**Comment by internetcreep** [comment]
[Provider Self-Input Form - Google Forms.pdf](https://github.com/radremedy/radremedy/files/421728/Provider.Self-Input.Form.-.Google.Forms.pdf)

It's not pretty, but the information is all here.

**Comment by alison985** [comment]
Thanks for the PDF! I added more fields: https://github.com/alison985/radremedy/commit/b65878bd2ee9f3a53898ad7995181451bf46f863 As I look at these questions some of them are location specific and it's clear you're handling the case of providers having multiple locations. From a data modeling perspective would it be worth it at this point to attach location-specific information to a location instead of to a provider?

**Comment by radioprotector** [comment]
Eventually, yes., but I feel like that would be a pretty significant rework.

In the short term, I feel like it's probably most appropriate to have resource entries for each individual location for a provider. The reason for that is the fact that different locations for the same provider often vary in terms of the services they offer, the languages spoken, and the staff working there. That last piece is most critical when we think about the review model.

At a minimum, we could clarify the address information to talk about *primary* locations.
